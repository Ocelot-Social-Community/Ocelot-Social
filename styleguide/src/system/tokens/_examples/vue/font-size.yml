#
# FONT-SIZE TOKENS
# Use these tokens to set a font-size.
#

props:
  - name: font-size-xxxx-large
    value: "3rem"
  - name: font-size-xxx-large
    value: "2.5rem"
  - name: font-size-xx-large
    value: "2rem"
  - name: font-size-x-large
    value: "1.5rem"
  - name: font-size-large
    value: "1.25rem"
  - name: font-size-base
    value: "1rem"
  - name: font-size-body
    value: "15px"
  - name: font-size-small
    value: "0.8rem"
  - name: font-size-x-small
    value: "0.7rem"
  - name: font-size-xx-small
    value: "0.6rem"
global:
  type: number
  category: font-size
