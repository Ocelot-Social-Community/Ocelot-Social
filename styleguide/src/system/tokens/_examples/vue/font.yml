#
# FONT TOKENS
# Use these tokens to set font-weight and font-family.
#

props:
  - name: font-family-heading
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-text
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-code
    value: "inconsolata, monospace"
    category: "font-family"

  - name: font-weight-regular
    value: "normal"
    category: "font-weight"
  - name: font-weight-bold
    value: "600"
    category: "font-weight"
global:
  type: ...
  category: font
