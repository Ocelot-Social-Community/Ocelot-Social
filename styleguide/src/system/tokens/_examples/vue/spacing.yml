#
# SPACING TOKENS
# Use these tokens to set padding, margin and position coordinates.
#

aliases:
  xxx-large:
    value: &xxx-large 120
  xx-large:
    value: &xx-large 60
  x-large:
    value: &x-large 40
  large:
    value: &large 28
  base:
    value: &base 20
  small:
    value: &small 12
  x-small:
    value: &x-small 6
  xx-small:
    value: &xx-small 4
  xxx-small:
    value: &xxx-small 2

props:
  - name: xxx-large
    value: *xxx-large
    category: space-size
  - name: xx-large
    value: *xx-large
    category: space-size
  - name: x-large
    value: *x-large
    category: space-size
  - name: large
    value: *large
    category: space-size
  - name: base
    value: *base
    category: space-size
  - name: small
    value: *small
    category: space-size
  - name: x-small
    value: *x-small
    category: space-size
  - name: xx-small
    value: *xx-small
    category: space-size
  - name: xxx-small
    value: *xxx-small
    category: space-size

  - name: space-xxx-large
    value: "{!xxx-large}px"
  - name: space-xx-large
    value: "{!xx-large}px"
  - name: space-x-large
    value: "{!x-large}px"
  - name: space-large
    value: "{!large}px"
  - name: space-base
    value: "{!base}px"
  - name: space-small
    value: "{!small}px"
  - name: space-x-small
    value: "{!x-small}px"
  - name: space-xx-small
    value: "{!xx-small}px"
  - name: space-xxx-small
    value: "{!xxx-small}px"

global:
  type: number
  category: space
