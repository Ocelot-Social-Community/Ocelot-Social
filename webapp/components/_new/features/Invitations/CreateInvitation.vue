<template>
  <div class="create-invitation">
    <div>{{ $t('invite-codes.generate-code-explanation') }}</div>
    <form @submit.prevent="generateInviteCode" class="generate-invite-code-form">
      <ds-input
        name="comment"
        :placeholder="$t('invite-codes.comment-placeholder')"
        v-model="comment"
        :schema="{ type: 'string', max: 30 }"
      />
      <base-button circle class="generate-invite-code" icon="plus" type="submit" />
    </form>
  </div>
</template>

<script>
export default {
  name: 'CreateInvitation',
  data() {
    return {
      comment: '',
    }
  },
  methods: {
    async generateInviteCode() {
      this.$emit('generate-invite-code', this.comment)
      this.comment = ''
    },
  },
}
</script>

<style scoped lang="scss">
.create-invitation {
  display: flex;
  flex-direction: column;
  gap: $space-small;
}

.generate-invite-code-form {
  display: flex;
  align-items: center;
  gap: $space-x-small;
}

::v-deep .ds-form-item {
  margin-bottom: 0;
  flex: auto;
}

::v-deep .ds-input-error {
  margin-top: $space-xx-small;
  margin-left: $space-x-small;
}
</style>
