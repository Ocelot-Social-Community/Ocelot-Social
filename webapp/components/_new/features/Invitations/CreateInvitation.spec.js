import { render, screen, fireEvent } from '@testing-library/vue'

import CreateInvitation from './CreateInvitation.vue'

const localVue = global.localVue

describe('CreateInvitation.vue', () => {
  let wrapper

  const Wrapper = ({ isDisabled = false }) => {
    return render(CreateInvitation, {
      localVue,
      propsData: {
        isDisabled,
      },
      mocks: {
        $t: jest.fn((v) => v),
      },
    })
  }

  it('renders', () => {
    wrapper = Wrapper({})
    expect(wrapper.container).toMatchSnapshot()
  })

  it('renders with disabled button', () => {
    wrapper = Wrapper({ isDisabled: true })
    expect(wrapper.container).toMatchSnapshot()
  })

  describe('when the form is submitted', () => {
    beforeEach(() => {
      wrapper = Wrapper({})
    })

    it('emits generate-invite-code with empty comment', async () => {
      const button = screen.getByRole('button')
      await fireEvent.click(button)
      expect(wrapper.emitted()['generate-invite-code']).toEqual([['']])
    })

    it('emits generate-invite-code with comment', async () => {
      const button = screen.getByRole('button')
      const input = screen.getByPlaceholderText('invite-codes.comment-placeholder')
      await fireEvent.update(input, 'Test comment')
      await fireEvent.click(button)
      expect(wrapper.emitted()['generate-invite-code']).toEqual([['Test comment']])
    })
  })
})
