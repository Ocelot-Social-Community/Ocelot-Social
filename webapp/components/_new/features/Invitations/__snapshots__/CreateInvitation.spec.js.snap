// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateInvitation.vue renders 1`] = `
<div>
  <div
    class="create-invitation"
  >
    <div>
      invite-codes.generate-code-explanation
    </div>
     
    <form
      class="generate-invite-code-form"
    >
      <div
        class="ds-form-item ds-input-size-base"
      >
        <label
          class="ds-input-label"
          style="display: none;"
        >
          
  

        </label>
        <div
          class="ds-input-wrap"
        >
          <!---->
          <input
            class="ds-input"
            name="comment"
            placeholder="invite-codes.comment-placeholder"
            tabindex="0"
            type="text"
          />
          <!---->
        </div>
        <transition-stub
          name="ds-input-error"
        >
          <div
            class="ds-input-error"
            style="display: none;"
          >
            
    
  
          </div>
        </transition-stub>
      </div>
       
      <button
        class="generate-invite-code base-button --icon-only --circle"
        type="submit"
      >
        <span
          class="base-icon"
        >
          <!---->
        </span>
         
        <!---->
         
      </button>
    </form>
  </div>
</div>
`;
